lib/Optimizer/GraphOptimizer.cpp

```diff
@@ -1794,6 +1794,11 @@ void glow::optimize(Function *F, CompilationMode mode) {
     optimizeQuantization(F);
   }
 
+  while (sinkCode(F)) {
```

Comment:
Let's say that sinkCode returns true. Then DCE inside the loop-body will be executed. Later, at some point sinkCode would return false and the loop would terminate. The last executed transformation performed by the will be DCE. And then after the loop you have another DCE call, which would be useless in such a case. 

To avoid this useless DCE call, you should probably place the loop after the last DCE call.

Reply:
good catch!